records/state/state_test.go

```diff
@@ -167,15 +192,35 @@ func state(st string) statusOpt {
 	return func(s *Status) { s.State = st }
 }
 
-func netinfo(ips ...string) statusOpt {
+func netinfos(netinfos ...NetworkInfo) statusOpt {
 	return func(s *Status) {
-		netinfos := &s.ContainerStatus.NetworkInfos
-		for _, ip := range ips {
-			*netinfos = append(*netinfos, NetworkInfo{IPAddress: ip})
-		}
+		s.ContainerStatus.NetworkInfos = append(s.ContainerStatus.NetworkInfos, netinfos...)
 	}
 }
 
+func netinfo(ips ...string) NetworkInfo {
+	netinfo := NetworkInfo{}
+	for _, ip := range ips {
+		netinfo.IPAddresses = append(netinfo.IPAddresses, IPAddress{ip})
+	}
+	return netinfo
+}
+
+// NetworkInfo using v0.25 syntax for storing a single IP.
+func oldnetinfo(ip string) NetworkInfo {
+	netinfo := NetworkInfo{}
+	netinfo.IPAddress = ip
+	return netinfo
+}
+
+// NetworkInfo using both 0.25 and 0.26 syntax for IPs.
+func oldnewnetinfo(oldip string, newip string) NetworkInfo {
```

Comment:
why? does mesos actually generate this? i thought there was no deprecation cycle between `ipaddress` and `ipaddresses` as far as mesos was concerned -- that it was a clean break.